v1/tarball/image_test.go

```diff
@@ -15,11 +15,48 @@
 package tarball
 
 import (
+	"archive/tar"
+	"bufio"
+	"io"
+	"os"
+	"path/filepath"
 	"testing"
 
 	"github.com/google/go-containerregistry/name"
 )
 
+func TestFlatten(t *testing.T) {
+	img, err := ImageFromPath("testdata/whiteout.tar", nil)
+	if err != nil {
+		t.Errorf("Error loading image: %v", err)
+	}
+	tarPath, _ := filepath.Abs("img.tar")
+	defer os.Remove(tarPath)
+	if err := Flatten(img, tarPath); err != nil {
+		t.Errorf("Error when flattening image: %v", err)
+	}
+	f, err := os.Open(tarPath)
+	if err != nil {
+		t.Errorf("Error when opening tar file for reading: %v", err)
+	}
+	r := bufio.NewReader(f)
```

Comment:
Don't need the bufio.Reader here either.

Reply:
:+1: